client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -158,7 +159,16 @@ private void handleLogSealed(Segment segment) {
                               * inflight that will need to be resent to the new segment when the write lock
                               * is released. (To preserve order)
                               */
-                             flushInternal();
+                             for (SegmentOutputStream writer : selector.getWriters()) {
+                                 try {
+                                     writer.write(PendingEvent.withoutHeader(null, ByteBufferUtils.EMPTY, null));
```

Comment:
Here, we are writing a zero byte append to all the segments when one of the segments receives a segment sealed response from the segment store. 
This approach implies that the EventStreamWriter would wait for the response from Segment store (a DataAppended , SegmentIsSealed) for each of the open segments. If a segment sealed is obeserved for any of the individual segment writers then EventStreamWriter would send zero byte appends to all the available segment writers again.

Also, does writing the zero byte appends to all the current segment writers before invoking a flush on all of the them help speed up things?

Reply:
No, it will be slower in the case where there are no inflight events. In the case that there are or that there are some in the buffer it won't change because the 0 byte event will be lumped into the open batch (taking up no space). Hence zero extra bytes will have to go over the wire.